### PR TITLE
Fix issue #26: [macOS] `DYLD_LIBRARY_PATH` env var needed for dev shell

### PR DESCRIPTION
**Pull request by yvan-sraka**
c.f. https://github.com/input-output-hk/cardano-node/pull/4952#issuecomment-1459312169

### CODE REVIEW

File: dynamic.nix [hunk 1]
```diff
@@ -24,7 +24,10 @@ pkgs.mkShell {
             ;;
         esac
         }
-    '';
+    '' ++ (if pkgs.stdenv.targetPlatform.isMacOS then ''
+    # this one is only needed on macOS right now, due to a bug in loading libcrypto.
+    export DYLD_LIBRARY_PATH=$(pkg-config --libs-only-L libcrypto|cut -c 3-)
+    '' else "");
```
Comment (angerman) [inline dynamic.nix:30]: can this be done with `pkgs.lib.optional`?

can't we just set `DYLD_LIBRARY_PATH = "${openssl.lib}/"` in nix though?

File: flake.nix [hunk 2]
```diff
@@ -68,19 +68,19 @@
...
             "aarch64-linux"
             "aarch64-darwin"
-       ];       
+       ];
     in flake-utils.lib.eachSystem supportedSystems (system:
-         let 
+         let
            pkgs = import nixpkgs {
-             overlays = [haskellNix.overlay] ++ __attrValues overlays;
+             overlays = [haskellNix.overlay] ++ builtins.attrValues overlays;
```
Comment (angerman) [inline flake.nix:83]: What's up with all of these __ -> builtins renaming? I honestly don't know which ones are better, __ was just shorter 🙄

**Review by angerman — CHANGES_REQUESTED** [review]
Let's try to not make that pkg-config --libs-only hack.

I still don't know why this isn't being picked up automatically, but I guess openssl is defines somewhere not as pkg-config dependency :(

File: flake.nix (hunk 2 above)
Comment (yvan-sraka) [reply flake.nix:83]: It's just that https://github.com/oxalica/nil complain that `__x` isn't defined :)

Comment (angerman) [reply flake.nix:83]: fair. I'd argue that's a bug in `nil`, but alright, that's a reason I can accept.

**Pull request merged**